Sources/NIOHTTPCompression/HTTPResponseDecompressor.swift

```diff
@@ -68,25 +86,27 @@ public final class NIOHTTPResponseDecompressor: ChannelDuplexHandler, RemovableC
 
             context.fireChannelRead(data)
         case .body(var part):
-            switch self.state {
-            case .compressed(_, let originalLength):
+            self.compression?.compressedLength += part.readableBytes
+            switch self.compression {
+            case .some(let compression):
                 while part.readableBytes > 0 {
+                    self.compression?.updateLength(with: part)
                     var buffer = context.channel.allocator.buffer(capacity: 16384)
                     do {
-                        try self.decompressor.decompress(part: &part, buffer: &buffer, originalLength: originalLength)
+                        try self.decompressor.decompress(part: &part, buffer: &buffer, compressedLength: compression.compressedLength)
```

Comment:
This block of code inconsistently uses `compression` and `self.compression?`. It'd be better to be a bit consistent on this front.